SCloudRandomizer/MainViewController.m

```diff
@@ -373,17 +378,31 @@ - (void)remoteControlReceivedWithEvent:(UIEvent *)event
     }
 }
 
+// Do something based on specific notifications
+- (void)receiveNotification:(NSNotification *) notification {
+    if ([[notification name] isEqualToString:@"StreamCompleted"]) {
+        NSLog (@"Recieved StreamCompleted notification!");
+        [self playNextTrack];
+    } else if ([[notification name] isEqualToString:@"ReadyToPlay"]) {
+        NSLog(@"Recieved ReadyToPlay notification!");
+        if (self.prepToPlayHud != nil) {
```

Comment:
yep it will be nil when pausing a song then resuming it. The hud is only used when playing for first time but the notification will happen everytime the stream plays.